src/test/kotlin/uk/gov/justice/digital/hmpps/educationandworkplanapi/app/database/jpa/JpaGoalPersistenceAdapterTest.kt

```diff
@@ -30,75 +32,139 @@ class JpaGoalPersistenceAdapterTest {
   @Mock
   private lateinit var goalMapper: GoalEntityMapper
 
-  @Test
-  fun `should create goal given action plan does not already exist`() {
-    // Given
-    val prisonNumber = aValidPrisonNumber()
-    val reference = UUID.randomUUID()
-    val domainGoal = aValidGoal(
-      reference = reference,
-    )
-
-    given(actionPlanRepository.findByPrisonNumber(any())).willReturn(null)
-
-    val entityGoal = aValidGoalEntity(
-      reference = reference,
-    )
-    given(goalMapper.fromDomainToEntity(any())).willReturn(entityGoal)
-
-    val actionPlanEntity = aValidActionPlanEntity(
-      prisonNumber = prisonNumber,
-      goals = mutableListOf(entityGoal),
-    )
-    given(actionPlanRepository.saveAndFlush(any<ActionPlanEntity>())).willReturn(actionPlanEntity)
-
-    given(goalMapper.fromEntityToDomain(any())).willReturn(domainGoal)
-
-    // When
-    val actual = persistenceAdapter.createGoal(domainGoal, prisonNumber)
-
-    // Then
-    assertThat(actual).isEqualTo(domainGoal)
-    verify(actionPlanRepository).findByPrisonNumber(prisonNumber)
-    verify(goalMapper).fromDomainToEntity(domainGoal)
-    verify(goalMapper).fromEntityToDomain(entityGoal)
+  @Nested
```

Comment:
A `@Nested` class is a bit like nesting `describe` functions in `jest` unit tests - it just allows us to group related tests 👍